lib/navigation-page.php

```diff
@@ -74,11 +74,43 @@ function gutenberg_navigation_init( $hook ) {
  * @return bool Filtered decision about loading block assets.
  */
 function gutenberg_navigation_editor_load_block_editor_scripts_and_styles( $is_block_editor_screen ) {
-	if ( is_callable( 'get_current_screen' ) && get_current_screen() && 'gutenberg_page_gutenberg-navigation' === get_current_screen()->base ) {
+	if ( is_navigation_editor_screen() ) {
 		return true;
 	}
 
 	return $is_block_editor_screen;
 }
 
 add_filter( 'should_load_block_editor_scripts_and_styles', 'gutenberg_navigation_editor_load_block_editor_scripts_and_styles' );
+
+/**
+ * Filters the Theme JSON to alter the settings of the Navigation block.
+ *
+ * @param Array $data the raw Theme JSON config at the Theme level.
+ * @return Array the amended Theme JSON config.
+ */
+function gutenberg_navigation_editor_filter_navigation_block_settings( $data ) {
+	if ( is_navigation_editor_screen() ) {
+		$data['settings']['blocks']['core/navigation'] = array(
+			'hasSubmenuIndicatorSetting'   => false,
+			'hasItemJustificationControls' => false,
+			'hasColorSettings'             => false,
```

Comment:
These x3 settings were originally props on the Navigation block. We has to use filters to override them. 

The new approach allow us to toggle these via Theme JSON.